pkg/autohealer/healer.go

```diff
@@ -428,3 +432,74 @@ func (h *Healer) updateSilence(e *common.Etcd) {
 		h.silences[sid] = &s
 	}
 }
+
+// syncSilencesFromBackend queries silences from the metric backend. Only Prometheus Alertmanager
+// is supported at this time.
+func (h *Healer) syncSilencesFromBackend(ctx context.Context, e *common.Etcd) {
+	if h.backend.GetType() != model.PrometheusType || !h.SyncSilences {
+		level.Info(h.logger).Log("msg", "Skip sync silences",
+			"backend", h.backend.GetAddress())
+		return
+	}
+	level.Info(h.logger).Log("msg", "Sync silences from metric backend",
+		"backend", h.backend.GetAddress())
+	silencesMap, err := h.backend.(*prometheus.Backend).GetAlertManagerSilences(ctx, []string{"instance=~\".+\""})
+	if err != nil {
+		level.Error(h.logger).Log("msg", "Error retrieving silences from Alertmanager",
+			"backend", h.backend.GetAddress())
+		return
+	}
+	for _, silence := range silencesMap {
+		// If silence's comment doesn't start with `[faythe]` prefix, ignore it.
+		if !strings.HasPrefix(strings.ToLower(*silence.Comment), "[faythe]") {
+			continue
+		}
+		s := &model.Silence{
+			Name:        "Auto sync silence from Alertmanger",
+			CreatedBy:   *silence.CreatedBy,
+			CreatedAt:   time.Time(*silence.StartsAt),
+			ExpiredAt:   time.Time(*silence.EndsAt),
+			Tags:        []string{"auto-sync", "alertmanager"},
+			Description: *silence.Comment,
+		}
+		for _, matcher := range silence.Matchers {
+			// Only care about matcher with name 'instance'
+			if *matcher.Name == "instance" {
+				s.Pattern = *matcher.Value
+				break
+			}
+		}
+		if s.Pattern == "" {
+			continue
+		}
+		if err := s.Validate(); err != nil {
+			level.Error(h.logger).Log("msg", "Error when validating silence", "err", err)
```

Comment:
is there any cases that silences is valid in AM and not valid here? that will throw a lot of errors.

Reply:
Not much, after TTL logic is updated. The Validate call is for generating ID and TTL.